Enhancement: Add `CONTRIBUTING.md` (#615)
php/web-php: .github/CONTRIBUTING.md

## Filing bugs

Bugs can be filed on [GitHub Issues](https://github.com/php/web-php/issues).

Where possible, please include a self-contained reproduction case!

## Feature requests

Feature requests can be filed on [GitHub Issues](https://github.com/php/web-php/issues).

## Writing tests

We love getting new tests! www.php.net is a fairly old project and improving test coverage is
a huge win for every www.php.net user.

[Our QA site includes a page detailing how to write test cases.](https://qa.php.net/write-test.php)

Submitting test scripts helps us to understand what functionality has changed.
It is important for the stability and maintainability of www.php.net that tests are
comprehensive.

## Getting help

If you are having trouble contributing to www.php.net, or just want to talk to a human
about what you're working on, you can contact us via the
[PHP php.net internal infrastructure discussion](mailto:php-webmaster@lists.php.net).

## Checklist for submitting contribution

- Update git source just before running your final `diff` and before testing.
- Create test scripts.
- Run

  ```

>   php tests/run-tests.php -q --show-diff

Author: Also see #621.

If your change is easy to review and obviously has no side-effects, it might be
committed relatively quickly.

Because www.php.net is a volunteer-driven effort, more complex changes will require
patience on your side. If you do not receive feedback in a few days, consider
bumping. Before doing this think about these questions:

- Did I send the patch to the right mailing list?
- Did I review the mailing list archives to see if these kind of changes had
  been discussed before?
- Did I explain my change clearly?
- Is my change too hard to review? If so, why?

## What happens when your contribution is applied?

Your name will likely be included in the Git commit log. 

## Git commit rules

This section refers to contributors that have Git push access and make commit
changes themselves. We'll assume you're basically familiar with Git, but feel
free to post your questions on the mailing list. Please have a look at the more
detailed [information on Git](https://git-scm.com/).

www.php.net is developed through the efforts of a large number of people. Collaboration
is a Good Thing(tm), and Git lets us do this. Thus, following some basic rules
with regards to Git usage will:

> - Make everybody happier, especially those responsible for maintaining PHP
>   itself.

Reviewer: ```suggestion
- Make everybody happier, especially those responsible for maintaining the website.
```

- Keep the changes consistently well documented and easily trackable.
- Prevent some of those 'Oops' moments.
- Increase the general level of good will on planet Earth.

Having said that, here are the organizational rules:

1. Respect other people working on the project.

> 2. Discuss any significant changes on the list before committing and get
>    confirmation from the release manager for the given branch.

Reviewer: Release manager does not apply here, so probably just:
```suggestion
2. Discuss any significant changes on the list before committing.
```

3. If you "strongly disagree" about something another person did, don't start
   fighting publicly - take it up in private email.

4. If you don't know how to do something, ask first!

5. Test your changes before committing them. We mean it. Really. To do so use

   ```

>    php tests/run-tests.php -q --show-diff`.

Author: Also see #621.

   ```

> Currently we have the following branches in use:
> 
> | Branch    |           |
> | --------- | --------- |
> | master    | Active development branch for PHP 8.2 |

Reviewer: I think this can be removed completely:
```suggestion
```

The next few rules are more of a technical nature:

1. Do not commit multiple files and dump all messages in one commit. If you
   modified several unrelated files, commit each group separately and provide a
   nice commit message for each one. See example below.

2. Do write your commit message in such a way that it makes sense even without
   the corresponding diff. One should be able to look at it, and immediately
   know what was modified. Definitely include the function name in the message
   as shown below.

3. In your commit messages, keep each line shorter than 80 characters. And try

>    to align your lines vertically, if they wrap. It looks bad otherwise.

Reviewer: Not sure what to do with this. There are mentions of "example below", although there is no example. It also talks about "include the function name in the message" which does not really apply to this codebase. Maybe remove this altogether, and just keep a note about using reasonable commit messages.

# CONTRIBUTING

We are using [GitHub Actions](https://github.com/features/actions) as a continuous integration system.

For details, take a look at the following workflow configuration file:

> - [`workflows/integrate.yaml`](workflows/integrate.yaml)

Author: Using a list here because we could add more workflows.

## Tests

We are using [`run-tests.php`](https://github.com/php/php-src/blob/f0c679c72ce02c1578ba9d56a099343b1eb3e16c/run-tests.php) to run [PHPT tests](https://qa.php.net/write-test.php).

Run

```

> php tests/run-tests.php

Reviewer 2: ```suggestion
php tests/run-tests.php -q --show-diff
```
